velox/dwio/common/IntDecoder.h

```diff
@@ -438,12 +441,41 @@ inline T IntDecoder<isSigned>::readInt() {
     return readLittleEndianFromBigEndian<T>();
   } else {
     if constexpr (std::is_same_v<T, int128_t>) {
+      if (numBytes_ == 12) {
+        VELOX_DCHECK(!useVInts_, "Int96 should not be VInt encoded.");
+        return readInt96();
+      }
       VELOX_NYI();
     }
     return readLongLE();
   }
 }
 
+template <bool isSigned>
+inline int128_t IntDecoder<isSigned>::readInt96() {
+  int64_t offset = 0;
+  unsigned char ch;
+
+  // Read 8 unsigned bytes.
+  uint64_t nanos = 0;
```

Comment:
Can we just do this?  This class should not have any knowledge about timestamp
```c++
int128_t result = 0;
for (int i = 0; i < 12; ++i) {
  auto ch = readByte();
  result |= static_cast<uint128_t>(ch & BASE_256_MASK) << (i * 8);
}
return result;
```
Then inside `TimestampColumnReader` we convert these `int128_t` into `Timestamp`, possible in-place.

Reply:
Updated following this suggestion. Now we convert `int128_t` to `Timestamp` in getValues inside `TimestampColumnReader`.  Please note that we also need to convert the `int128_t` value as `Timestamp` in `TimestampRange`  with this change. Thanks.

Reply:
`IntDecoder` and `TimestampColumnReader` looks good now, the change in `TimestampRange` is not ideal though, that is parquet specific logic we need to find a way to move this into parquet as well.